scipy/linalg/lapack.py

```diff
@@ -499,9 +499,43 @@ def get_lapack_funcs(names, arrays=(), dtype=None):
 
     In LAPACK, the naming convention is that all functions start with a
     type prefix, which depends on the type of the principal
-    matrix. These can be one of {'s', 'd', 'c', 'z'} for the numpy
-    types {float32, float64, complex64, complex128} respectevely, and
-    are stored in attribute `typecode` of the returned functions.
+    matrix. These can be one of {`'s'`, `'d'`, `'c'`, `'z'`} for the numpy
+    types {float32, float64, complex64, complex128} respectively, and
+    are stored in attribute ``typecode`` of the returned functions.
+
+    Examples
+    --------
+    Suppose we would like to use '?lange' routine which computes the selected
+    norm of an array. We pass our array in order to get the correct 'lange'
+    flavor.
+
+    >>> from scipy.linalg import lapack as lap
+    >>> a = np.random.rand(3,2)
+    >>> x_lange = lap.get_lapack_funcs('lange', (a,))
+    >>> x_lange
+    <fortran dlange>
+    >>> x_lange = lap.get_lapack_funcs('lange',(a*1j,))
+    >>> x_lange
+    <fortran zlange>
+
+    Several LAPACK routines work best when its internal WORK array has
+    the optimal size (big enough for fast computation and small enough to
+    avoid waste of memory). This size is determined also by a dedicated query
+    to the function which is often wrapped as a standalone function and
+    commonly denoted as ``###_lwork``. Below is an example for ``?sysv``
+
+    >>> from scipy.linalg import lapack as lap
+    >>> a = np.random.rand(1000,1000)
+    >>> b = np.random.rand(1000,1)*1j
+    >>> # We pick up zsysv and zsysv_lwork due to b array
+    ... xsysv, xlwork = lap.get_lapack_funcs(('sysv', 'sysv_lwork'), (a, b))
+    >>> opt_lwork = lap._compute_lwork(xlwork, a.shape[0])
```

Comment:
Should I create a public frontend to this? Because you have been promoting this to use in a couple of PRs as far as I remember. Since this is a low level function I think people who oughta know this should also know that single/double culprit.